command_line/rescale_diamond_anvil_cell.py

```diff
@@ -0,0 +1,302 @@
+        # by the anvils.  This is an array of shape (N).
+        transmission = np.exp(-linear_atten_coeff * l0)
+        transmission *= np.exp(-linear_atten_coeff * l1)
+        # Remove redundant things that scale with N.
+        del l0, l1
+
+        # Correct the measured intensities for this attenuation.
+        for col in "intensity.prf.value", "intensity.sum.value":
+            try:
+                rtable[col].set_selected(sel, refls[col] / flex.double(transmission))
+            except KeyError:
+                pass
+        # Correct the measured variances accordingly.
+        for col in "intensity.prf.variance", "intensity.sum.variance":
+            try:
+                rtable[col].set_selected(
+                    sel, refls[col] / flex.double(np.square(transmission))
+                )
+            except KeyError:
+                pass
+
+
+def run(args=None, phil=phil_scope):  # type: (List[str], libtbx.phil.scope) -> None
+    """
+    Run dials.rescale_diamond_anvil_cell as from the command line.
+
+    Take integrated experiment lists and reflection tables and correct the all the
+    integrated intensities for the estimated attenuation by the diamond anvils.
+
+    Args:
+        args: The arguments supplied by the user (default: sys.argv[1:]).
+        phil: The PHIL scope definition (default: phil_scope, the master PHIL scope
+              for this program).
+    """
+    usage = "dials.%s [options] integrated.expt integrated.refl" % filename_root
+
+    parser = OptionParser(
+        usage=usage,
+        phil=phil,
+        read_reflections=True,
+        read_experiments=True,
+        check_format=False,
+        epilog=__doc__,
+    )
+
+    params, options = parser.parse_args(args=args, show_diff_phil=False)
+
+    # Configure the logging.
+    dials.util.log.config(options.verbose, logfile=params.output.log)
+
+    # Log the difference between the PHIL scope definition and the active PHIL scope,
+    # which will include the parsed user inputs.
+    diff_phil = parser.diff_phil.as_str()
+    if diff_phil:
+        logger.info("The following parameters have been modified:\n%s", diff_phil)
+
+    # Check that at least one reflection table and experiment list have been provided.
+    input_errors = []
+    if not params.input.experiments:
+        input_errors.append(
+            "Please provide at least one valid experiment list (.expt) file."
+        )
+    if not params.input.reflections:
+        input_errors.append(
+            "Please provide at least one valid reflection table (.refl) file."
+        )
+    if input_errors:
+        sys.exit("\n".join([parser.format_help()] + input_errors))
+
+    # These functions are commonly used to collate the input.
+    experiments = flatten_experiments(params.input.experiments)
+    reflections_list = flatten_reflections(params.input.reflections)
+    reflections = reflections_list[0]
```

Comment:
Note, if you wanted to follow my comment and split the input reflections_list into a list of single-dataset tables, just use `parse_multiple_datasets` from `dials.util.multi_dataset_handling`

Reply:
Hi @jbeilstenedmands, I've reformatted this to use tools from `dials.util.multi_dataset_handling` as per your suggestion.  Could you look over it and verify that I'm holding it correctly?